cmd/amp/cli/cli_test.go

```diff
@@ -72,12 +85,17 @@ func loadFiles(t *testing.T) []*TestSpec {
 }
 
 func parseCmd(t *testing.T, test *TestSpec) {
+	var sortedCmdSpecs []CommandSpec
 	commandMap, err := generateCmdSpec(test.contents)
 	if err != nil {
 		t.Error(err)
 		return
 	}
-	for _, cmdSpec := range commandMap {
+	for _, k := range commandMap {
+		sortedCmdSpecs = append(sortedCmdSpecs, k)
+	}
+	sort.Sort(CommandSpecByStep(sortedCmdSpecs))
```

Comment:
I think `MapSlice`s would do the job of keeping the order of the map. Check https://godoc.org/gopkg.in/yaml.v2#MapSlice